### PR TITLE
assaneRed_Fix_SCS_testcases

### PR DESCRIPTION
**Pull request by assaneRed**
Fix SCS testcases by adding error code 403 for the first http request

### CODE REVIEW

File: src/harness/testcases/WINNF_FT_S_SCS_testcase.py [hunk 1]
```diff
@@ -202,7 +202,7 @@ def test_WINNF_FT_S_SCS_12(self,config_filename):
     Checks that SAS UUT response with fatal alert message.
     """
     config = loadConfig(config_filename)
-    self.assertTlsHandshakeFailure(client_cert=config['clientCert'],
+    self.assertTlsHandshakeOrHttpsFailure(client_cert=config['clientCert'],
                                    client_key=config['clientKey'])
 
   @winnforum_testcase
```
[the remark this answers is not in the document]
Comment (assaneRed) [reply src/harness/testcases/WINNF_FT_S_SCS_testcase.py]: done

File: src/harness/security_testcase.py [hunk 2]
```diff
@@ -247,11 +247,29 @@ def _InfoCb(conn, where, ok):
...
+      self.assertTlsHandshakeFailure(client_cert, client_key)
+    except AssertionError as e:
+      try:
+        device_a = json.load(
+          open(os.path.join('testcases', 'testdata', 'device_a.json')))
+        request = {'registrationRequest': [device_a]}
+        response = self._sas.Registration(request, ssl_cert=client_cert,
+                                          ssl_key=client_key)['registrationResponse']
+      except AssertionError as e:
+        self.assertEqual(e.args[0], 403)
```
[the remark this answers is not in the document]
Comment (assaneRed) [reply src/harness/security_testcase.py]: the behavior is determined in the file request_handler.py:

except pycurl.error as e:
    raise AssertionError(e.args[0], e.args[1])
  http_code = conn.getinfo(pycurl.HTTP_CODE)
  conn.close()
  body = response.getvalue()
  logging.info('Response:\n' + body)
  assert http_code == 200, http_code

and pycurl.error doesn't have an error code 403. 
so e.arg[0] will be 403 only if it is http code error, right?

File: src/harness/testcases/WINNF_FT_S_SCS_testcase.py [hunk 3]
```diff
@@ -170,7 +170,7 @@ def test_WINNF_FT_S_SCS_10(self,config_filename):
     """
     config = loadConfig(config_filename)
     try:
-      self.assertTlsHandshakeFailure(client_cert=config['clientCert'],
+      self.assertTlsHandshakeOrHttpsFailure(client_cert=config['clientCert'],
```
[the remark this answers is not in the document]
Comment (assaneRed) [reply src/harness/testcases/WINNF_FT_S_SCS_testcase.py]: - Yes you have right. Whe don't need SCS.10

- we need to check them, but I think it is better to submit it or to submit this PR and then update them.

File: src/harness/security_testcase.py [hunk 4]
```diff
@@ -266,6 +266,7 @@ def assertTlsHandshakeOrHttpsFailure(self, client_cert=None, client_key=None):
         response = self._sas.Registration(request, ssl_cert=client_cert,
                                           ssl_key=client_key)['registrationResponse']
       except AssertionError as e:
+        logging.debug('HTTPS forbidden: 403')
```
Comment (kate-harrison) [inline src/harness/security_testcase.py]: This logs what you *hope* has happened, but not necessarily what did happen. Maybe something like "TLS session established, expecting HTTP error 403; received HTTP error %d"?

Comment (assaneRed) [reply src/harness/security_testcase.py]: Yes you have right. thanks

**Comment by assaneRed** [comment]
Hi Paul,
Can you check, if you have time, and let me know if any others comments.
Thanks

**Comment by assaneRed** [comment]
Hi Paul,
I have change SCS_testcase based on your suggestions, can you check again, if you have time, and let me know if any others comments.
Thanks

**Comment by assaneRed** [comment]
Hello Paul,
Can you check again, please

Best regard,
Assane

File: src/harness/request_handler.py [hunk 5]
```diff
@@ -112,6 +112,13 @@ def _Request(url, request, config, is_post_method):
...
+
+  if http_code != 200:
+    raise HTTPError(http_code)
   if body:
     return json.loads(body.decode('utf-8'))
+
+class HTTPError(Exception):
+  def __init__(self, error_code):
+    self.error_code  = error_code
+    super(HTTPError, self).__init__( error_code)
```
Comment (pmarks-net) [inline src/harness/request_handler.py]: Delete space before `error_code`

File: src/harness/request_handler.py [hunk 6]
```diff
@@ -112,6 +112,13 @@ def _Request(url, request, config, is_post_method):
...
-  assert http_code == 200, http_code
+
+  if http_code != 200:
+    raise HTTPError(http_code)
   if body:
     return json.loads(body.decode('utf-8'))
+
+class HTTPError(Exception):
+  def __init__(self, error_code):
+    self.error_code  = error_code
```
Comment (pmarks-net) [inline src/harness/request_handler.py]: Delete extra space before `=`

**Pull request merged**